packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts

```diff
@@ -51,10 +53,24 @@ export class SamplerOperations {
 
     constructor(
         protected readonly _samplerContract: ERC20BridgeSamplerContract,
-        public readonly bancorService?: BancorService,
+        public readonly provider?: SupportedProvider,
         public readonly balancerPoolsCache: BalancerPoolsCache = new BalancerPoolsCache(),
+        protected readonly getBancorServiceFn?: () => BancorService,
```

Comment:
This is just for testing dependency injection, right?

Reply:
Yep!